Develop

flassagne opened the pull request:
Handle Qml modules with qtcreator

Commit:
feat(qtcreator): add an export qml function

Commit:
feat(cmut/qtcreator): add target qml in import path to use inspector and code completion

Commit:
feat(install): install qml target and export in QML_LOCATION

Commit:
refactor(qtcreator): remove export function use QML_LOCATION in import function

Commit:
refactor(cmut): use a private cmut target option to install and import Qml

ledocc commented on install/cmut__install__install_qml.cmake:
```diff
@@ -0,0 +1,42 @@
+             PATTERN "*.qml"
+             PATTERN "qmldir" )
+
+     set_target_properties(${target} PROPERTIES CMUT__QML_DIRECTORIES ${ARG__DESTINATION})
```
CMUT__QML_DIRECTORIES, this is a list, so get previous list, add element, set list

flassagne replied:
Done

ledocc commented on install/cmut__install__install_qml.cmake:
```diff
@@ -0,0 +1,42 @@
+        ${ARGN}
+    )
+
+    if(NOT ARG__DIRECTORY)
```
this one could not have default value

flassagne replied:
Done

flassagne replied:
Done

ledocc commented on install/cmut__install__install_qml.cmake:
```diff
@@ -0,0 +1,42 @@
+     set_target_properties(${target} PROPERTIES CMUT__QML_DIRECTORIES ${ARG__DESTINATION})
+
+     get_target_property(_export_properties ${target} EXPORT_PROPERTIES)
+     list(APPEND _export_properties CMUT__QML_DIRECTORIES)
```
test if CMUT__QML_DIRECTORIES is not already in EXPORT_PROPERTIES
```
if(NOT CMUT__QML_DIRECTORIES IN_LIST EXPORT_PROPERTIES)
...
endif()
```

flassagne replied:
Done

Commit:
fix(install): make argument DIRECTORY mandatory and append not override CMUT__QML_DIRECTORIES

Commit:
refactor(qtcreator): add optional import qml target

Commit:
fix(install): do not append directory NOTFOUND

Commit:
fix(qtcreator): accept CMUT__QML_DIRECTORIES contain an list

ledocc commented on qtcreator/cmut__qtcreator__import_qml.cmake:
```diff
@@ -0,0 +1,58 @@
+# PREDICATE Library directory is one delph children of install directory
+
+    if( ${CMAKE_BUILD_TYPE} STREQUAL "Release" )
+         get_target_property(_dependency_location ${dependency} IMPORTED_LOCATION_RELEASE)
```
try the current config, then try  the default config
```
string(TOUPPER ${CMAKE_BUILD_TYPE} BUILD_TYPE_UPPER)
get_target_property(_dependency_location ${dependency} IMPORTED_LOCATION_${CMAKE_BUILD_TYPE})
if(NOT _dependency_location)
    get_target_property(_dependency_location ${dependency} IMPORTED_LOCATION)
endif()
```

flassagne replied:
Done

ledocc commented on qtcreator/cmut__qtcreator__import_qml.cmake:
```diff
@@ -0,0 +1,58 @@
+            continue()
+        endif()
+
+        __cmut__qtcreator__get_dependency_install_directory(_dependency_install_dir ${_dependency})
```
test if _dependency_install_dir exists
```
if(EXISTS /the/path/to/test)
    ...
endif()
```

flassagne replied:
done

ledocc commented on qtcreator/cmut__qtcreator__import_qml.cmake:
```diff
@@ -0,0 +1,58 @@
+    endif()
+
+    if( NOT _dependency_location )
+        cmut_error("We can't find your dependency location. Verify your build type (Release, Debug, Coverage)")
```
log like this:
```
cmut_error("[cmut][qtcreator][import_qml] - ...")
```

flassagne replied:
Done

ledocc commented on qtcreator/cmut__qtcreator__import_qml.cmake:
```diff
@@ -0,0 +1,58 @@
+
+    __cmut__qtcreator__get_qml_import_paths(_qml_directories "${_target_dependencies}")
+
+    set( QML_IMPORT_PATH "${_qml_directories}" CACHE STRING "Qml import path" FORCE)
```
set QML_IMPORT_PATH as INTERNAL data to not see it in cmake gui, thought?

flassagne replied:
Done

Commit:
feat: add a CMakeLists.txt in order to load cmut project

Commit:
fix(qtcreator): make QML_IMPORT_PATH internal and verify dependency install dir

Commit:
fix(install): append CMUT__QML_DIRECTORIES in export properties just it needed

ledocc commented on install/cmut__install__install_qml.cmake:
```diff
@@ -0,0 +1,64 @@
+
+function(__cmut__export_qml_directories target)
+
+    if(NOT CMUT__QML_DIRECTORIES IN LIST EXPORT_PROPERTIES)
```
IN_LIST

flassagne replied:
Done

Commit:
fix(install): use the right keywork IN_LIST

Commit:
fix(qtcreator): import QML_IMPORT_PATH correctly

- find imported location
- fix keyword
- append qml import path in cmake cache variable
